### ISSUE TITLE
Differences in FOV calculation depending on position

### ISSUE THREAD

**Issue by scalemaildev** [ondras/rot.js#218]
I've noticed that the FoV calculation puts out different results based on the source position. For example, in these two images: standing on the bones in the lower left allows the player to see the hex next to the bones in the upper right. However, standing in that same spot in the second image doesn't allow the player to see their original position.

![cornercase1](https://github.com/ondras/rot.js/assets/15042849/978c9db5-2882-4341-8313-9b80f83cc0d7)
![cornercase2](https://github.com/ondras/rot.js/assets/15042849/8279ca96-8ccc-4c55-ade5-9f0b1cce5c2c)

Is this normal behavior? I'm using flat-topped hexes with double wide coordinates, and the topology is set to 6.

Code snippet for how I'm implementing fov:

```
updateVis(scene: any, entity: Entity): void {
        const vis = new Set<number>();
        const viewPosC = intToCoord(entity.c.e.pos);
        const px = viewPosC[0];
        const py = viewPosC[1];

        const fov = new FOV.PreciseShadowcasting((x: number, y: number): boolean => {
            const pos = coordToInt([ x, y ]);
            return scene.currentLevel.tiles[pos] && !scene.currentLevel.tiles[pos].c.e.blocksSight;
        }, { topology: 6 });

        fov.compute(px, py, entity.c.mem.visRadius, (xi: number, yi: number) => {
            const coord = coordToInt([ xi, yi ]);
            if (scene.currentLevel.tiles[coord]) vis.add(coord);
        });

        entity.c.mem.vis = vis;
    }
```

**Comment by ondras** [comment]
Yeah, I am afraid this is just the way it is. The algorithm is not *symmetric*, i.e. "you see B from A" does not imply "you see A from B".

I understand that this might not be ideal for some games, but I am not aware of any straightforward way (algorithm modification) to enforce the rule.

Visibility in general is complex and hard. See https://www.reddit.com/r/Damnthatsinteresting/comments/1agtkaj/optimal_distance_to_peek_in_shooter_games/ for an interesting example.
